var/spack/repos/builtin/packages/casacore/package.py

```diff
@@ -63,15 +66,18 @@ class Casacore(CMakePackage):
     depends_on("fftw@3.0.0: precision=float,double", when="~fftpack")
     depends_on("sofa-c", type="test")
     depends_on("hdf5", when="+hdf5")
-    depends_on("adios2+mpi", when="+adios2")
+    depends_on("adios2@2.6.0:", when="+adios2")
+    depends_on("adios2+mpi", when="+adios2+mpi")
+    depends_on("adios2~mpi", when="+adios2~mpi")
     depends_on("mpi", when="+mpi")
-    depends_on("python@2.6:", when="+python")
+    depends_on("python@2.6:", when="@:3.5.0 +python")
+    depends_on("python@3:", when="@3.6.0: +python")
     depends_on("boost +python", when="+python")
     depends_on("boost +system +filesystem", when="+dysco")
-    depends_on("py-numpy", when="+python")
+    depends_on("py-numpy@:1.99.99", when="@:3.6.0 +python")
```

Comment:
If the aim is to restrict to just numpy 1, then the following is preferred.
```suggestion
    depends_on("py-numpy@:1", when="@:3.6.0 +python")
```
Your version would allow for `py-numpy@:1.99.100` if that is what you need in this case (I imagine it isn't).

Reply:
done